File: apps/sepuling-java/src/main/java/smecalculus/bezmen/storage/SepulkaDao.java
package smecalculus.bezmen.storage;

import java.util.Optional;
import java.util.UUID;
import smecalculus.bezmen.core.SepulkaStateDm;

/**
 * Port: server side
 */
public interface SepulkaDao {
    SepulkaStateDm.AggregateRoot add(SepulkaStateDm.AggregateRoot state);

    Optional<SepulkaStateDm.Existence> getBy(String externalId);

    Optional<SepulkaStateDm.Preview> getBy(UUID internalId);

    void updateBy(UUID internalId, SepulkaStateDm.Touch state);
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/bezmen/storage/SepulkaDao.kt
package smecalculus.bezmen.storage

import smecalculus.bezmen.core.SepulkaStateDm
import java.util.UUID

/**
 * Port: server side
 */
interface SepulkaDao {
    fun add(state: SepulkaStateDm.AggregateRoot): SepulkaStateDm.AggregateRoot

    fun getBy(externalId: String): SepulkaStateDm.Existence?

    fun getBy(internalId: UUID): SepulkaStateDm.Preview?

    fun updateBy(
        internalId: UUID,
        state: SepulkaStateDm.Touch,
    )
}
